Node/intelligence-SpeechToText/app.js

```diff
@@ -30,35 +30,37 @@ const connector = new builder.ChatConnector({
 });
 
 const bot = new builder.UniversalBot(connector);
-server.post("/api/messages", connector.listen());
+server.post('/api/messages', connector.listen());
 
 //=========================================================
 // Bots Dialogs
 //=========================================================
 
-bot.dialog("/", session => {
+bot.dialog('/', session => {
     if (hasAudioAttachment(session)) {
         var stream = getAudioStreamFromAttachment(session.message.attachments[0]);
         speechService.getTextFromAudioStream(stream)
             .then(text => {
-                session.send(processText(session.message.text, text));
+                session.send(processText(text));
             })
             .catch(error => {
-                session.send("Oops! Something went wrong. Try again later.");
+                session.send('Oops! Something went wrong. Try again later.');
                 console.error(error);
             });
-    }
-    else {
-        session.send("Did you upload an audio file? I'm more of an audible person. Try sending me a wav file");
+    } else {
+        session.send('Did you upload an audio file? I\'m more of an audible person. Try sending me a wav file');
     }
 });
 
 //=========================================================
 // Utilities
 //=========================================================
 const hasAudioAttachment = session => {
-    return ((session.message.attachments.length > 0) && ((session.message.attachments[0].contentType === "audio/wav") || (session.message.attachments[0].contentType === "application/octet-stream")));
-}
+    return session.message.attachments.length > 0 &&
+        (session.message.attachments[0].contentType.indexOf('audio') === 0 ||
+         session.message.attachments[0].contentType === 'video/mp4' || // FB sends audio streams as MP4
```

Comment:
Even though the code now allows processing of video/mp4 and audio/aac file formats as used by Facebook audio recordings and attachments, the Bing Speech API cannot handle these file formats and an error is returned.